src/common/decorators/storage.ts

```diff
@@ -86,49 +88,70 @@ class Storage {
       }
     } catch (err: any) {
       // Safari in private mode doesn't allow localstorage
+    } finally {
+      if (this._listeners[storageKey]) {
+        this._listeners[storageKey].forEach((listener) =>
+          listener(oldValue, value)
+        );
+      }
     }
   }
 }
 
-const subscribeStorage = new Storage();
+const storages: Record<string, StorageClass> = {};
 
-export const LocalStorage =
-  (
-    storageKey?: string,
-    property?: boolean,
-    subscribe = true,
-    storageType?: globalThis.Storage,
-    propertyOptions?: PropertyDeclaration
-  ): any =>
+export const Storage =
+  (options: {
+    storageKey?: string;
+    property?: boolean;
+    subscribe?: boolean;
+    storage?: globalThis.Storage;
```

Comment:
```suggestion
    storage?: "localStorage", "sessionStorage"
```
Because if we allow other storages, it will not be stored in `storages`.

Reply:
I think that is fine, only subscribe will not work?